### PR TITLE
Remove simple_poa consensus

### PR DESCRIPTION
**Pull request by Byeongjee**
It resolves #111 
I removed simple_poa consensus since it's no longer needed.
Saluki chain which exploits simple_poa engine is also removed.

### CODE REVIEW

**Comment by majecty** [comment]
Looks good to me. Let's wait the CI tests.

**Comment by majecty** [comment]
Please use full sentences in the commit description.

**Comment by junha1** [comment]
Good job

**Comment by Byeongjee** [comment]
@sgkim126  @majecty  I changed a lot.
I removed force/mandatory resealing, and some trace messages which seem to be wrong.
Please review them carefully.

File: core/src/client/client.rs [hunk 1]
```diff
@@ -76,7 +76,7 @@ pub struct Client {
 
     importer: Importer,
 
-    /// Timer for reseal_min_period/reseal_max_period on miner client
```
Comment (sgkim126) [inline core/src/client/client.rs]: AFAIR, The tendermint consensus also uses reseal_max_period to prepare the situation there is no transactions in the block.
@majecty Would you check it?

Comment (majecty) [reply core/src/client/client.rs]: Tendermint consensus always creates a block even though there is no transaction. So it is safe to remove it.

FYI, there is a separate timer in Tendermint code to sleep a few seconds when creating an empty block. However, the timer will be removed after the Tenderand is merged.

**Comment by Byeongjee** [comment]
@majecty Travis CI test passed successfully on my local machine.

**Comment by majecty** [comment]
@Byeongjee Please resolve the conflict.

**Pull request merged**